### PR TITLE
Add processType to crash ping schema

### PR DESCRIPTION
**Pull request by akkomar**

### CODE REVIEW

File: templates/telemetry/crash/crash.4.schema.json [hunk 1]
```diff
@@ -70,6 +70,10 @@
             }
           }
         },
+        "processType": {
+          "type": "string",
+          "enum": [ "main", "content", "gpu" ]
```
Comment (mreid-moz) [inline templates/telemetry/crash/crash.4.schema.json]: Do we want the ping to fail validation if we introduce a new `processType`?  If there's no strong reason to make this an `enum`, I would somewhat prefer to leave this as a free-form string value.

Comment (akkomar) [reply templates/telemetry/crash/crash.4.schema.json]: Good point, I'll leave it as free-form string.

Comment (georgf) [reply templates/telemetry/crash/crash.4.schema.json]: +1, a free-form string value would be more flexible in the future.
Is there any potential down-side to allowing free-form values, e.g. would it affect analysis in some way if unexpected values show up?

Comment (mreid-moz) [reply templates/telemetry/crash/crash.4.schema.json]: I checked some of the existing data and it is well-behaved (one of these three values or null), so I'm more worried about intentionally-introduced new values (maybe `dynamic`?) than a bunch of unexpected values.

If we start to observe unexpected values here it seems like we should address it at that time.

Comment (akkomar) [reply templates/telemetry/crash/crash.4.schema.json]: But how would you know you're getting unexpected values if it's not validated?
On the other hand I totally agree with the flexibility point.

Comment (mreid-moz) [reply templates/telemetry/crash/crash.4.schema.json]: You'd start to see them appear in datasets / analyses downstream.

Comment (fbertsch) [reply templates/telemetry/crash/crash.4.schema.json]: I'm kind of the opinion that we should have just included all current known process types [0]. But as per usual I'm late to the party :)

[0] https://searchfox.org/mozilla-central/source/toolkit/components/crashes/nsICrashService.idl#24

Comment (mreid-moz) [reply templates/telemetry/crash/crash.4.schema.json]: I'm not confident enough in our alerting on validation failures yet, so I'm inclined to be permissive / future-proof where possible.

Comment (fbertsch) [reply templates/telemetry/crash/crash.4.schema.json]: @mreid-moz that is concerning, can you make a bug to track that?

Comment (fbertsch) [reply templates/telemetry/crash/crash.4.schema.json]: I think we should prioritize that bug highly so we don't have to be concerned with missing data, and ending up with too lenient schemas.

**Pull request merged**